Source/WebCore/html/parser/HTMLConstructionSite.cpp

```diff
@@ -641,7 +641,8 @@ void HTMLConstructionSite::insertTextNode(const String& characters, WhitespaceMo
         if (UNLIKELY(breakIndex == currentPosition))
             breakIndex = characters.length();
 
-        auto substring = characters.substring(currentPosition, breakIndex - currentPosition);
+        unsigned substringLength = breakIndex - currentPosition;
+        auto substring = LIKELY(!currentPosition && substringLength >= characters.length()) ? characters : characters.substring(currentPosition, substringLength);
```

Comment:
Either before or after this change, can we try moving this change into the inline part of `substring()` itself? I suspect `substring()` is **often** used when it’s likely the entire string. Maybe doing it globally would hurt things, but I am guessing that it would help more than it would hurt. Might even speed up Speedometer more.

Reply:
That's a good idea. I'll try in a follow-up since this patch already went through perf A/B testing.